### PR TITLE
Update the Contribution Guidelines

### PR DESCRIPTION
**Pull request by viklund**
### Describe the pull request:
- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [x] Documentation change
- [ ] Language translation

### Pull request long description:
This is a suggestion of updates to the contributing guidelines. Mostly it's information about how to work with git, pull requests and code review. Surprisingly little is changed compared to the old guidelines.

I also removed some outdated sections related to ZenHub and Agile project management. We could flesh those things out in this PR, but I haven't done that yet.

### CODE REVIEW

**Comment by silverdaz** [comment]
Please update the link that is inside `/docs` that was pointing to the `CONTRIBUTING.rst` file (for the readthedocs.org).
That's also the reason why it was a .rst file and not .md (eventhough I prefer .md)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+Slack channel. You can alternatively pick one already created.
+
+Assign yourself to that issue.
+
+Discussions on how to proceed about that issue take place in the comment
+section on that issue, beforehand.
+
+The keyword here is *beforehand*. It is usually a good idea to talk about it
+first. Somebody might have already some pieces in place, we avoid unnecessary
+work duplication and a waste of time and effort.
```
Comment (blankdots) [inline CONTRIBUTING.md]:
```suggestion
work duplication, _surprise_ features and a waste of time and effort.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Other people may also look at and review the code.
+
+A reviewers job is to:
+
+  * Read the code and make sure it is understandable
+  * Make sure that commit messages and commits are structured so that it is
+    possible to understand why certain changes were made.
+  * Ensure that the test-suite covers the new behavior
+
```
Comment (blankdots) [inline CONTRIBUTING.md]: > If there are change requests, a reviewer should provide constructive feedback relevant to the issue addressed by the Pull Request.

I strongly believe we could do better with the feedback we provide on reviews.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+Assign yourself to that issue.
+
+Discussions on how to proceed about that issue take place in the comment
+section on that issue, beforehand.
+
+The keyword here is *beforehand*. It is usually a good idea to talk about it
+first. Somebody might have already some pieces in place, we avoid unnecessary
+work duplication and a waste of time and effort.
+
```
Comment (blankdots) [inline CONTRIBUTING.md]: > Once work on a feature/bug has been completed, make sure to provide helpful commit messages and create Pull Request. If a Pull Request tries to solve more than **one** issue it should probably be split into several PRs. 

I thinking if we can make exception for bugs and allow a PR to contain more than **one** bug fix.

Comment (blankdots) [inline CONTRIBUTING.md]: > Once work on a feature/bug has been completed, make sure to provide helpful commit messages and create Pull Request. If a Pull Request tries to solve more than **one** issue it should probably be split into several PRs. 

I am thinking if we can make exception for bugs and allow a PR to contain more than **one** bug fix.

File: CONTRIBUTING.md (hunk 2 above)
Comment (viklund) [reply CONTRIBUTING.md]: Do you mean better than the standard we are currently at? Yes, we can probably do that.

File: CONTRIBUTING.md (hunk 3 above)
Comment (viklund) [reply CONTRIBUTING.md]: Personally, I prefer several small ones to a few big ones. But as long as we don't reject pull requests because they are "too small" for some reason I am happy. I think it should be fine with a one-line pull request if it actually fixes a current bug or missing feature.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -39,96 +39,104 @@ Every day at 10:30 we have standups in Zoom meeting: <id>
...
+and also makes the code easier to review.
+
+Write helpful commit messages that describes the changes and possibly why they
+were necessary.
+
+Each commit should contain changes that are functionally connected and/or
+related. If you for example want to write _and_ in a commit message this is a
+good sign that it should have been two commits.
+
+Learn how to select chunkgs of changed files to do multiple separate commits of
```
Comment (norling) [inline CONTRIBUTING.md]: spelling: "chunks"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -39,96 +39,104 @@ Every day at 10:30 we have standups in Zoom meeting: <id>
...
+unrelated things. This can be done with either `git add -p` or `git commit -p`.
 
-   Use [git rebase -i] in
-   order to rewrite your history, and have meaningful commits.  That
-   way, we avoid the _'Typo'_, _'Work In Progress (WIP)'_, or
-   _'Oops...forgot this or that'_ commits.
 
-   Limit the first line of your git commits to 72 characters or less.
+#### Helpfull commit messages
```
Comment (norling) [inline CONTRIBUTING.md]: I think we should use "helpful" instead of "helpfull". It's not technically incorrect, but "helpful" is the more common spelling.

File: CONTRIBUTING.md (hunk 3 above)
Comment (blankdots) [reply CONTRIBUTING.md]: I also prefer small PRs as they are usually easier to follow, but I assume the discussion is not if it is "too small" or not (it can be a small change with big impact), but if it is of value to the team and the project.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+first. Somebody might have already some pieces in place, we avoid unnecessary
+work duplication and a waste of time and effort.
+
+
+## How we work with Git
+
+All work take place in feature branches. Give your branch a short descriptive
+name and prefix the name with:
+
+   * `feature/` if it is a code feature
```
Comment (dtitov) [inline CONTRIBUTING.md]: Do we need a `test/` prefix to work on tests? Or it goes to the `feature/`-prefixed branches?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+### How we do code reviews
+
+A code review is initiated by making a Pull Request in the appropriate repo on
+github.
+
+Work should not continue on the branch _unless_ it has been marked with the `DO
+NOT MERGE` tag. Once the PR is ready and the `DO NOT MERGE` tag is removed the
+review process can go on to the next stage.
+
+Before review starts it is a good idea to rebase your branch to `master` to
```
Comment (dtitov) [inline CONTRIBUTING.md]: Is it "a good idea" or is it actually required? I'm personally for a "good idea" approach :)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+Once all the reviews are positive the Pull Request can be _merged_ into
+`master` and the feature branch deleted.
+
+
+----
+
+Thanks again,
+/NBIS System Developers
+
+[searching under Issues]: https://github.com/NBISweden/LocalEGA/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Abug%20%5BBUG%5D%20in%3Atitle
```
Comment (blankdots) [inline CONTRIBUTING.md]:
```suggestion
[searching under Issues]: https://github.com/EGA-archive/LocalEGA/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Abug%20%5BBUG%5D%20in%3Atitle
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+[open a new one]: https://github.com/NBISweden/LocalEGA/issues/new?title=%5BBUG%5D
```
Comment (blankdots) [inline CONTRIBUTING.md]:
```suggestion
[open a new one]: https://github.com/EGA-archive/LocalEGA/issues/new?title=%5BBUG%5D
```

File: CONTRIBUTING.md (hunk 7 above)
Comment (viklund) [reply CONTRIBUTING.md]: Well, if you rebase before, the test-suite will test what the merge will look like instead of what a merge would have looked like at the graft point. If the branch is rebased before merge conflicts can be resolved before the PR is done as well. For those reasons I think required would be better.

File: CONTRIBUTING.md (hunk 6 above)
Comment (viklund) [reply CONTRIBUTING.md]: I think extending test suite is `feature/`. But now that you mention this, we have no prefix for just plain bugfix (as in, not urgent). Maybe we should have a `bugfix/` prefix, what do other people think?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+test-suite is for.
+
+Once all the reviews are positive the Pull Request can be _merged_ into
+`master` and the feature branch deleted.
+
+
+----
+
+Thanks again,
+/NBIS System Developers
```
Comment (blankdots) [inline CONTRIBUTING.md]: Probably this could be something else as well e.g. 
> LocalEGA development team

File: CONTRIBUTING.md (hunk 6 above)
Comment (norling) [reply CONTRIBUTING.md]: The angular guidelines have quite a few [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type), and while I don't think we need this many I'd like a set like **docs**, **feature**, **bugfix**, **test**, and maybe **refactor**. I might be overly pedantic, but to me it's descriptive and is easier since there are fewer edge cases with the categories. :)

File: CONTRIBUTING.md (hunk 10 above)
Comment (viklund) [reply CONTRIBUTING.md]: Indeed. It should be something like this.

File: CONTRIBUTING.md (hunk 6 above)
Comment (viklund) [reply CONTRIBUTING.md]: Well, there are several ways of skinning a cat. I don't think I like the **refactor** one, since a refactor should be part of a feature in general (though it should be a separate set of commits). I think it's enough with **bugfix** and **feature** for now, where everything that is not a **bugfix** is a **feature**. We can't really have **hotfixes** yet as there's no production system running.

File: CONTRIBUTING.md (hunk 7 above)
Comment (viklund) [reply CONTRIBUTING.md]: Just to add to my previous comment. Both of these activities I mention is work that has to be done, and it's the original developer that knows best how to resolve conflicts. So not rebasing is basically just a way of moving work from now into the future. And if there's a lot of merge conflicts, at least parts of the review will have to be redone resulting in double-work.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+Before review starts it is a good idea to rebase your branch to `master` to
+ensure that we can make a clean merge.
+
+The initiator of the PR should recruit 2 reviewers that get assigned reviewer
+duty on the branch. It's not a bad idea to announce this on the #localega slack
+channel.
+
+Other people may also look at and review the code.
+
+A reviewers job is to:
```
Comment (viklund) [inline CONTRIBUTING.md]: I think we should add something here along the lines of

```
 * Make constructive, helpful non-categorical comments.
```

An invented example of the kind of behaviour I would like to avoid is something like this:

> This variable has the wrong name.

Instead it should be:

> I think the intent of this code could be clearer if the variable had a different name

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+Every day at 10:30 we have standups in Zoom meeting: <id>
+
+### Sprint Reviews and Planning
+
+Approximately every 2 weeks we have sprint reviews and then also a sprint
+planning meeting.
+
+
+
+## How to work on a new feauture/bug
```
Comment (omllobet) [inline CONTRIBUTING.md]: "feauture" typo

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+#### Helpfull commit messages
+
+The commit messages may be seen as meta-comments on the code that are
+incredibly helpful for anyone who wants to know how this piece of software is
+working, including colleagues (current and future) and external users.
+
+Some tips about writing helpful commit messages:
+
+#. Separate subject (the first line of the message) from body with a blank line.
```
Comment (omllobet) [inline CONTRIBUTING.md]: On https://github.com/EGA-archive/LocalEGA/blob/feature/git-guidelines/CONTRIBUTING.md I see all these messages on the same line but I am not sure if readthedocs markdown is different from github.

File: CONTRIBUTING.md (hunk 8 above)
Comment (silverdaz) [reply CONTRIBUTING.md]: I would also change the "NBIS System Developers" to "EGA System Developers" too.

File: CONTRIBUTING.md (hunk 13 above)
Comment (viklund) [reply CONTRIBUTING.md]: That's a bug. Will fix this.

File: CONTRIBUTING.md (hunk 6 above)
Comment (blankdots) [reply CONTRIBUTING.md]: I don't think we need such a complicated branching scheme. We can make use of additional labels to distinguish what type of `/feature` it is.

Comment (dtitov) [reply CONTRIBUTING.md]: I personally support Martin's idea, I like this branching sheme: **docs**, **feature**, **bugfix**, **test**, and **refactor**.

File: CONTRIBUTING.md (hunk 3 above)
Comment (nanjiangshu) [reply CONTRIBUTING.md]: A bug can be big and a feature can be small. What should be the upper limit size of one (suitable) PR can result in endless discussions. I also prefer small PRs and agree with @blankdots that such statement should be added. I think an issue of feature/bug should be able to be described by one or a few sentences.

File: CONTRIBUTING.md (hunk 11 above)
Comment (nanjiangshu) [reply CONTRIBUTING.md]: `A reviewers job` should be changed to `A reviewer's job`

File: CONTRIBUTING.md (hunk 8 above)
Comment (juhtornr) [reply CONTRIBUTING.md]: Should we do something for that repo under NBIS organisation, e.g. archive it?

File: CONTRIBUTING.md (hunk 12 above)
Comment (juhtornr) [reply CONTRIBUTING.md]:
```suggestion
## How to work on a new feature/bug
```
